src/nvim/ex_cmds.c

```diff
@@ -1032,8 +1032,10 @@ void do_bang(int addr_count, exarg_T *eap, bool forceit, bool do_in, bool do_out
   }
 
   if (bangredo) {  // put cmd in redo buffer for ! command
-    if (!prevcmd_is_set()) {
```

Comment:
what about the other `prevcmd_is_set` case? can that be usefully handled with this new feature, or it doesn't make sense?

Reply:
If I understood properly the other (former) case, it involves also the `:!` and not the `:{range}!`. Hence, if we still want the error `No previous command` on user input `:!!` we should not change it, but I may very possibly be wrong.